### PR TITLE
150247828 handle doit create dup

### PR DESCRIPTION
**Pull request by Maroo-b**
ticket: https://www.pivotaltracker.com/story/show/150247828

- Store id of created Doit volunteer op via LocalSupport services
- Verify the existence of Doit volunteer op in trace history before
importing

### CODE REVIEW

**Comment by tansaku** [comment]
great work @Maroo-b - thanks so much

Is there any chance you could add some cuke tests to cover this at the acceptance level?  I get the sense you're not a cucumber acceptance level test fan :-)

also there's a couple of code climate issues to check if you have a moment ...

**Comment by Maroo-b** [comment]
@tansaku I updated the PR.

**Comment by tansaku** [comment]
great stuff @Maroo-b 

Two thoughts:

1) the doit id in the cuke could be pulled into an internal instance variable to make it more readable?
2) given the op has been posted is the link to the doit op stored and accessible?  Particularly if we're not importing it, it would be good to give the admin the link to the doit op they posted so they could navigate to it (new feature?)

**Comment by Maroo-b** [comment]
@tansaku 
1- the doit id is retrieved manually from the VCR cassette file, I guess that exposing the value is easier for modification in case the cassette file was updated, what do you think?
2- The link isn't stored currently, only the new generated id from doit platform. It seems that's a new feature to implement. (For example where to display the stored Doit vol op link? )

File: app/services/doit/post_to_doit.rb [hunk 1]
```diff
@@ -19,7 +19,8 @@ def call
       options[:headers] = {'X-API-Key' => DOIT_AUTH_TOKEN,
                            'Content-Type' => 'application/json'}
       options[:body] = build_request_body
-      http_adapter.post("#{ENV['DOIT_HOST']}#{doit_volonteer_op_resource}", options)
+      doit_res = http_adapter.post(doit_volonteer_op_resource, options)
+      JSON.parse(doit_res.body)['data']['opportunity']['id']
```
Comment (Maroo-b) [inline app/services/doit/post_to_doit.rb]: @tansaku the `call` was originally a command method, but to store the generated it now parse and return the id, do you consider it as a violation of command-query separation? I tried to make the return optional via blocks but I didn't manage to achieve it.

**Comment by tansaku** [comment]
> 1- the doit id is retrieved manually from the VCR cassette file, I guess that exposing the value is easier for modification in case the cassette file was updated, what do you think?

I take your point - I guess we have the tension between the test exposing something that's important to the developer and exposing the intention on the part of the non-technical admin.  Would it be any less easy to modify if the implementation detail of the id was in the step definition?  The thing that I'm looking for in cucumber scenarios is a high level description that's going to have the best chance of being understood by someone new to the project, but of course that always has to be balanced against making sure that whoever is going to manipulate the steps has a great idea of what's going on under the hood ... tricky ...

> 2- The link isn't stored currently, only the new generated id from doit platform. It seems that's a new feature to implement. (For example where to display the stored Doit vol op link? )

yeah, new feature sensibly I guess - but an important one for the admin of the charity - the link should be just displayed in the show page for that volop - because we're keeping it local and not importing the remote one right?  The link will just be adding that id to the correct URL stem right?

File: app/services/doit/post_to_doit.rb (hunk 1 above)
Comment (tansaku) [reply app/services/doit/post_to_doit.rb]: ah hmm, I'll have to look more closely at this ...

**Comment by tansaku** [comment]
@Maroo-b if I don't hear back from you soon I might need to re-assign this or possibly take this on myself - hope you're well :-)

**Comment by tansaku** [comment]
@Maroo-b or we could pair on it if you follow me on twitter: https://www.agileventures.org/premium-f2f-offer ....?

**Comment by Maroo-b** [comment]
@tansaku sorry for the late reply, but is it possible to merge this PR and create a new one for displaying doit link in show page?

**Pull request merged**